aiida/manage/configuration/profile.py

```diff
@@ -127,33 +128,43 @@ def __init__(self, name, attributes, from_config=False):
         # Currently, whether a profile is a test profile is solely determined by its name starting with 'test_'
         self._test_profile = bool(self.name.startswith('test_'))
 
-    def get_repository_container(self) -> 'Container':
-        """Return the container of the profile's file repository.
+    @property
+    def _repository_container_path(self) -> str:
+        """Return the path to the container of the profile file repository."""
+        return os.path.join(self.repository_path, 'container')
 
-        :return: the profile's file repository container.
-        """
+    def _get_repository_container(self) -> 'Container':
+        """Return the disk object store ``Container``."""
         from disk_objectstore import Container
+        return Container(self._repository_container_path)
 
-        filepath = self._container_path
-        container = Container(filepath)
+    def get_repository(self) -> 'Repository':
+        """Return the UUID that uniquely identifies the repository configured for this profile."""
+        from aiida.repository import Repository
+        from aiida.repository.backend import DiskObjectStoreRepositoryBackend
+        container = self._get_repository_container()
+        self.initialise_repository()
+        backend = DiskObjectStoreRepositoryBackend(container=container)
+        return Repository(backend=backend)
 
-        if not self.container_is_initialised:
-            container.init_container(clear=True, **self.defaults['repository'])  # pylint: disable=unsubscriptable-object
+    def get_repository_uuid(self) -> Optional[str]:
+        """Return the UUID that uniquely identifies the repository configured for this profile.
+
+        .. note:: will return ``None`` if the repository has not yet been initialized.
```

Comment:
```suggestion
        .. note:: will return ``None`` if the repository has not yet been initialised.
```
Else @chrisjsewell will be angry. 😁